pkg/features/kube_features.go

```diff
@@ -334,6 +334,16 @@ const (
 	// deletion ordering.
 	HonorPVReclaimPolicy featuregate.Feature = "HonorPVReclaimPolicy"
 
+	// owner: @aojea
+	// Deprecated: v1.29
+	//
+	// Changes when the Node addresses are assigned to the Node when using
+	// external cloud provider. The default is for only the external cloud provider
+	// to set the Node addresses. Enabling this means that the kubelet, if not address
+	// has been configured, will assign addresses to the Node based on the --node-ip
```

Comment:
The `--node-ip` is still used by the external CCM via the `alpha.kubernetes.io/provided-node-ip` annotation though AFAIK. I might be worth clarifying this as its not straight forward (e.g. what occurs when InitializeNodeAddressesCloudProviderExternal set to true and false in the presence or absense of a `--node-ip` flag).

Reply:
The annotation is a hint only 

https://github.com/kubernetes/kubernetes/blob/e8abe1af8dcb36f65ef7aa7135d4664b3db90e89/pkg/kubelet/nodestatus/setters.go#L98-L116

the annotated address will not be added if does not match the cloud provider addresses

https://github.com/kubernetes/kubernetes/blob/e8abe1af8dcb36f65ef7aa7135d4664b3db90e89/staging/src/k8s.io/cloud-provider/node/helpers/address.go#L131-L134